reason-react-native/src/apis/PushNotificationIOS.md

```diff
@@ -0,0 +1,203 @@
+---
+id: apis/PushNotificationIOS
+title: PushNotificationIOS
+wip: true
+---
+
+```reason
+module Notification = {
+  type t;
+
+  [@bs.send] [@bs.return nullable]
+  external getAlert: t => option(Js.Json.t) = "";
+
+  [@bs.send] [@bs.return nullable]
+  external getMessage: t => option(Js.Json.t) = "";
+
+  [@bs.send] [@bs.return nullable]
+  external getSound: t => option(string) = "";
+
+  [@bs.send] [@bs.return nullable]
+  external getCategory: t => option(string) = "";
+
+  // Actually type ContentAvailable = 1 | null | void ...
+  [@bs.send] external getContentAvailable: t => bool = "";
+
+  [@bs.send] [@bs.return nullable]
+  external getBadgeCount: t => option(int) = "";
+
+  [@bs.send] [@bs.return nullable]
+  external getData: t => option(Js.Json.t) = "";
+
+  [@bs.send] [@bs.return nullable]
+  external getThreadID: t => option(string) = "";
+};
+
+type localNotification;
+
+[@bs.obj]
+external localNotification:
+  (
+    ~alertBody: string=?,
+    ~alertTitle: string=?,
+    ~alertAction: string=?,
+    ~soundName: string=?,
+    ~isSilent: bool=?,
+    ~category: string=?,
+    ~userInfo: Js.Json.t=?,
+    ~applicationIconBadgeNumber: int=?,
+    ~fireDate: Js.Date.t=?,
+    ~repeatInterval: [@bs.string] [
+                       | `minute
+                       | `hour
+                       | `day
+                       | `week
+                       | `month
+                       | `year
+                     ]
+                       =?,
+    unit
+  ) =>
+  localNotification =
+  "";
+
+[@bs.module "react-native"] [@bs.scope "PushNotificationIOS"]
+external presentLocalNotification: localNotification => unit = "";
+
+[@bs.module "react-native"] [@bs.scope "PushNotificationIOS"]
+external scheduleLocalNotification: localNotification => unit = "";
+
+[@bs.module "react-native"] [@bs.scope "PushNotificationIOS"]
+external cancelAllLocalNotifications: unit => unit = "";
+
+[@bs.module "react-native"] [@bs.scope "PushNotificationIOS"]
+external removeAllDeliveredNotifications: unit => unit = "";
+
+type deliveredNotification = {
+  .
+  "identifier": string,
+  "date": Js.Nullable.t(string),
+  "title": Js.Nullable.t(string),
+  "body": Js.Nullable.t(string),
+  "category": Js.Nullable.t(string),
+  "thread-id": Js.Nullable.t(string),
+  "userInfo": Js.Nullable.t(Js.Json.t),
+};
+
+[@bs.module "react-native"] [@bs.scope "PushNotificationIOS"]
+external getDeliveredNotifications:
+  (array(deliveredNotification) => unit) => unit =
+  "";
+
+[@bs.module "react-native"] [@bs.scope "PushNotificationIOS"]
+external removeDeliveredNotifications: (~identifiers: array(string)) => unit =
+  "";
```

Comment:
`~identifiers` isn't necessary here, did you include that to hint what argument the function requires?

Reply:
Yes, exactly. I was a bit hesitant about this, too, but otherwise it is not really clear what this `array(string)` really is.